lib/reline/line_editor.rb

```diff
@@ -708,10 +708,15 @@ def add_dialog_proc(name, p, context = nil)
     # Clear and rerender all dialogs line by line
     Reline::IOGate.hide_cursor
     ymin, ymax = (ranges_to_restore.keys + new_dialog_ranges.keys).minmax
+    screen_y_range = (@scroll_partial_screen || 0)..(@scroll_partial_screen || 0) + @screen_height - 1
```

Comment:
I wasn't familiar with range values' precedence so I confused a bit. I think make it a bit easier to understand:

```suggestion
    scroll_partial_screen = @scroll_partial_screen || 0
    screen_y_range = scroll_partial_screen..(scroll_partial_screen + @screen_height - 1)
```